src/wire/mod.rs

```diff
@@ -174,6 +176,13 @@ pub use self::icmpv6::{Message as Icmpv6Message,
                        ParamProblem as Icmpv6ParamProblem,
                        Packet as Icmpv6Packet,
                        Repr as Icmpv6Repr};
+#[cfg(any(feature = "proto-ipv4", feature = "proto-ipv6"))]
+pub use self::icmp::{Message as IcmpMessage,
+                     DstUnreachable as IcmpDstUnreachable,
+                     TimeExceeded as IcmpTimeExceeded,
+                     ParamProblem as IcmpParamProblem,
+                     Packet as IcmpPacket,
+                     Repr as IcmpRepr};
```

Comment:
Are you planning to somehow unify the ICMPv4 and ICMPv6 `Message`, `DstUnreachable`, `TimeExceeded` and `ParamProblem` enums? This doesn't seem like the right way to go to me. Why is this necessary?

Reply:
Indeed, it isn't